src/state/hooks.ts

```diff
@@ -0,0 +1,21 @@
+import { TypedUseSelectorHook, useSelector, ReactReduxContext } from 'react-redux';
+import type { RootState, AppDispatch } from './store';
+import { useContext } from 'react';
+
+const noopDispatch: AppDispatch = (): undefined => {};
+
+export const useAppDispatch = (): AppDispatch => {
```

Comment:
The direct imports from `react-redux` throw when used outside of Redux context which is usually great but if we want to continue to have our individual components to work outside of Chart context (like the Area example we discussed earlier) then we need alternative hooks that return undefined instead of throwing.

Reply:
Later we'll continue to have a re-exported AppDispatch anyways for typing purposes so this is the right track